FordLabs/peoplemover
# Update packages to remove requirement of using --legacy-peer-deps during install

## Issue
Resolves N/A

When cloning the repo and installing for the first time, the npm install fails in the UI folder currently, and you're forced to use the `--legacy-peer-deps` flag. This is less than ideal, so this PR resolved that issue. You should now be able to delete the node modules folder and install without any problem.

## What was done
- [x] Updated [react-datepicker](https://reactdatepicker.com/) from 2.14.1 to 4.8.0
- [x] Removed [react-text-mask](https://github.com/text-mask/text-mask) which is no longer maintained and swapped it for [react-input-mask](https://www.npmjs.com/package/react-input-mask)
- [x] Removed `@types/moment-timezone` as it is no longer needed
- [x]  Removed `eslint-plugin-standard` as it is no longer needed
- [x] Removed `jest-dom` and replaced it with `@testing-library/jest-dom` per npm's suggestion

## How to test
1. While in the UI folder, delete your node modules folder and npm install. Ensure you can install fully without using the `--legacy-peer-deps` flag.
2. Ensure selecting a start and end date works as expected on the product form
3. Ensure all tests pass and the build is successful.
